azure/azure.go

```diff
@@ -326,6 +326,57 @@ func (a *azureOps) DeleteFrom(diskName, _ string) error {
 	return a.Delete(diskName)
 }
 
+func (a *azureOps) Expand(
+	diskName string,
+	newSizeInGiB uint64,
+) (uint64, error) {
+	disk, err := a.disksClient.Get(
+		context.Background(),
+		a.resourceGroupName,
+		diskName,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	if disk.DiskProperties == nil || disk.DiskProperties.DiskSizeGB == nil {
+		return 0, fmt.Errorf("disk properties of (%v) is nil", diskName)
+	}
+
+	if *disk.DiskProperties.DiskSizeGB >= int32(newSizeInGiB) {
+		// no work is needed
+		return uint64(*disk.DiskProperties.DiskSizeGB), nil
+	}
+	oldSizeInGiB := uint64(*disk.DiskProperties.DiskSizeGB)
+	// Azure resizes in chunks of GiB even if the disk properties variable is DiskSizeGB
+	newSizeInGiBInt32 := int32(newSizeInGiB)
+	disk.DiskProperties.DiskSizeGB = &newSizeInGiBInt32
+
+	ctx := context.Background()
+	future, err := a.disksClient.CreateOrUpdate(
```

Comment:
nit: Just call Update here instead of CreateOrUpdate. Although it doesn't matter because you are doing a Get before so it won't create a new disk anyway.